lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -306,13 +318,23 @@ impl<'a> FuncGen<'a> {
         for i in begin..end {
             self.trap_table.offset_to_code.insert(i, code);
         }
+        self.instructions_address_map.push(InstructionAddressMap {
+            srcloc: SourceLoc::new(self.src_loc),
+            code_offset: begin,
+            code_len: end - begin,
+        });
```

Comment:
It seems that InstructionAddressMap is redundant with TrapTable. If trap table changed to take a length as well, we could just update its contents once and then produce an InstructionAddressMap at the end storing the correct data.

Reply:
Oh, I was deliberate on this. I wanted to separate the address map from the traps itself. It's the same design as cranelift has. The main reason for doing it this way is because it enforces better structure over all (since we might need to add `InstructionAddressMap` apart from traps in the future)

Reply:
OK! If they logically contain different data but just happen to include the same data now because of a partial implementation that's needed now, that's fine then. Thanks for the clarification!